macros/blueprint-proc-macro/src/benchmark.rs

```diff
@@ -0,0 +1,79 @@
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+use syn::parse::{Parse, ParseStream};
+use syn::{ItemFn, Token};
+
+// Defines custom keywords
+mod kw {
+    syn::custom_keyword!(threads);
+    syn::custom_keyword!(job_id);
+}
+
+/// `BenchmarkArgs` is a struct that holds the arguments for the `benchmark` macro.
+pub(crate) struct BenchmarkArgs {
+    /// The number of threads this benchmark should run with.
+    ///
+    /// `#[benchmark(threads = 4)]`
+    threads: syn::LitInt,
```

Comment:
I think there should be an option that tests your machine for how many threads it can reasonably support, then benchmarks with threads `1-num_threads`

Reply:
This way we can get metrics for a whole breadth of machine specs, no?